src/main/java/de/retest/ui/descriptors/DefaultRetestIdProvider.java

```diff
@@ -17,23 +16,20 @@ public String getRetestId( final IdentifyingAttributes identifyingAttributes ) {
 			throw new NullPointerException( "IdentifyingAttributes must not be null." );
 		}
 
-		// order is "text", "type"
 		final String text = identifyingAttributes.get( "text" );
-		if ( text != null ) {
-			return makeUnique( cut( normalizeString( text ) ) );
-		}
-
 		final String type = identifyingAttributes.get( "type" );
-		return makeUnique( cut( normalizeString( type ) ) );
+		final String rawId = text != null ? text : type;
+		return makeUnique( rawId );
```

Comment:
I think it is not clear that `makeUnique` subsumes `cut` and `normalize`. I'd rather call them separately.